signer/tests/integration/transaction_coordinator.rs

```diff
@@ -1791,6 +1784,672 @@ async fn sign_bitcoin_transaction() {
+    // Step 11 - Make two proper deposits
+    // -------------------------------------------------------------------------
+    // - Use the UTXOs confirmed in steps (5) and (10) to construct two
+    //   proper deposit request transactions. Submit them to the bitcoin
+    //   network and then inform Emily.
+    // =========================================================================
+    // Now lets make a deposit transaction and submit it
+    let utxo = depositor2.get_utxos(rpc, None).pop().unwrap();
+
+    let amount = 3_500_000;
+    let signers_public_key2 = shares2.aggregate_key.into();
+    let max_fee = amount / 2;
+    let (deposit_tx, deposit_request, _) =
+        make_deposit_request(&depositor2, amount, utxo, max_fee, signers_public_key2);
+    rpc.send_raw_transaction(&deposit_tx).unwrap();
+
+    let body = deposit_request.as_emily_request();
+    deposit_api::create_deposit(emily_client.config(), body)
+        .await
+        .unwrap();
+
+    let utxo = depositor1.get_utxos(rpc, None).pop().unwrap();
+    let amount = 4_500_000;
+    let signers_public_key1 = shares1.aggregate_key.into();
+    let max_fee = amount / 2;
+    let (deposit_tx, deposit_request, _) =
+        make_deposit_request(&depositor1, amount, utxo, max_fee, signers_public_key1);
+    rpc.send_raw_transaction(&deposit_tx).unwrap();
+
+    let body = deposit_request.as_emily_request();
+    deposit_api::create_deposit(emily_client.config(), body)
+        .await
+        .unwrap();
+
+    // =========================================================================
+    // Step 12 - Confirm the deposit and wait for the signers to do their
+    //           job.
+    // -------------------------------------------------------------------------
+    // - Confirm the deposit request. This will trigger the block observer
+    //   to reach out to Emily about deposits. It will have two so the
+    //   signers should do basic validations and store the deposit request.
+    // - Each TxSigner process should vote on the deposit request and
+    //   submit the votes to each other.
+    // - The coordinator should submit a sweep transaction. We check the
+    //   mempool for its existence.
+    // =========================================================================
+    faucet.generate_blocks(1);
+
+    wait_for_signers(&signers).await;
+
+    let (ctx, _, _, _) = signers.first().unwrap();
+    let mut txids = ctx.bitcoin_client.inner_client().get_raw_mempool().unwrap();
+
+    assert_eq!(txids.len(), 1);
+
+    let block_hash = faucet.generate_blocks(1).pop().unwrap();
+
+    wait_for_signers(&signers).await;
+
+    // =========================================================================
+    // Step 13 - Assertions
+    // -------------------------------------------------------------------------
+    // - During each bitcoin block, the signers should sign and broadcast a
```

Comment:
You could probably toss one in the db directly if you wanted it to be more "real" and less workarounds

Reply:
Hmmm what do you mean?